packages/compass-aggregations/src/modules/pipeline.js

```diff
@@ -307,6 +313,18 @@ const toggleStageCollapse = (state, action) => {
   return newState;
 };
 
+const checkIsMissingAtlasOnlyStageSupport = (newState, index, error) => (
+  newState.env !== ADL &&
+  newState.env !== ATLAS &&
+  ([SEARCH, SEARCH_META, DOCUMENTS].includes(newState[index].stageOperator)) ||
```

Comment:
Is the `||` here intentional? It seems like this would render the rest of the expression unnecessary